cxx/isce3/geometry/Topo.cpp

```diff
@@ -179,59 +180,76 @@ topo(Raster & demRaster, TopoLayers & layers)
         // Allocate vector for storing satellite position for each line
         std::vector<Vec3> satPosition(blockLength);
 
-        // For each line in block
-        double tline;
-        for (size_t blockLine = 0; blockLine < blockLength; ++blockLine) {
-
-            if (blockLine % std::max((int) (blockLength / 100), 1) == 0)
-                printf("\rTopo progress (block %d/%d): %d%%",
-                       (int) block + 1, (int) nBlocks,
-                       (int) (blockLine * 1e2 / blockLength)),
-                       fflush(stdout);
-
-            // Global line index
-            size_t line = lineStart + blockLine;
-
-            // Initialize orbital data for this azimuth line
-            Basis TCNbasis;
-            Vec3 pos, vel;
-            _initAzimuthLine(line, tline, pos, vel, TCNbasis);
-
-            satPosition[blockLine] = pos;
-
-            // Compute velocity magnitude
-            const double satVmag = vel.norm();
-
-            // For each slant range bin
-            #pragma omp parallel for reduction(+:totalconv)
-            for (size_t rbin = 0; rbin < _radarGrid.width(); ++rbin) {
-
-                // Get current slant range
-                const double rng = _radarGrid.slantRange(rbin);
-
-                // Get current Doppler value
-                const double dopfact = (0.5 * _radarGrid.wavelength()
-                                     * (_doppler.eval(tline, rng) / satVmag)) * rng;
-
-                // Store slant range bin data in Pixel
-                Pixel pixel(rng, dopfact, rbin);
-
-                // Initialize LLH to middle of input DEM and average height
-                Vec3 llh = demInterp.midLonLat();
-
-                // Perform rdr->geo iterations
-                int geostat = rdr2geo(
-                    pixel, TCNbasis, pos, vel, _ellipsoid, demInterp, llh,
-                    _radarGrid.lookSide(), _threshold, _numiter, _extraiter);
-                totalconv += geostat;
-
-                // Save data in output arrays
-                _setOutputTopoLayers(llh, layers, blockLine, pixel, pos, vel,
-                        TCNbasis, demInterp);
+        // Get the midpoint of the DEM block in LLH coordinates.
+        // This should always be safe since the call to `computeDEMBounds()`
+        // above loads a block of the DEM raster.
+        assert(demInterp.hasRaster());
```

Comment:
```suggestion
        assert(demInterp.haveRaster());
```
Interesting. Looks like this bug hasn't been caught by our internal CI.

Reply:
@rtburns-jpl those Debug build CI job are paying off! :wink: